dcos/dcos-ui
# DCOS-47536: get package image from nested items

Get package images for icons from services that are inside folders.

Closes DCOS-47536.

## Testing

This is a weird one to test because icons don't appear in development. To test this I did:

- Apply @TattdCodeMonkey changes from #3536 . After this you will still see placeholder icons because the image URLs we are getting for packages look like `https://localhost:4200/package/resource?url=https://downloads.mesosphere.com/assets/universe/000/cassandra-icon-small.png` (why??)
- To get around that, change `ServicesTable` `getImage` (line 209) to
```
    // Get service image/icon
    const imgSrc = service.getImages()["icon-small"].split("=")[1];
    return (
      <span className="icon icon-mini icon-image-container icon-app-container icon-margin-right">
        <img src={imgSrc} />
      </span>
    );
```
- Go to catalog > cassandra > review & run > change `name` to `somefolder/cassandra`
- Go to services & click on `somefolder`: you should see correct icon beside cassandra

## Trade-offs

Not that I'm aware of

## Dependencies 

None
